### PR TITLE
Add cmake support

### PR DESCRIPTION
**Pull request by jberkel**
ready to be reviewed

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  ${SRC_DIR}/high_resolution_clock.cpp
+  ${SRC_DIR}/http_server.cpp
+  ${SRC_DIR}/queue.cpp
+  ${SRC_DIR}/simulator.cpp
+  ${SRC_DIR}/tcp_socket.cpp
+)
+
+add_library(simulator ${SIMULATOR_SRC_FILES})
+include_directories(${Boost_INCLUDE_DIR} ${INCLUDE_DIR})
+set(CMAKE_CXX_FLAGS "-std=c++11")
```
Comment (arvidn) [inline CMakeLists.txt]: is this really how you're supposed to do this? won't this break the build for anything other than clang and gcc?

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  ${SRC_DIR}/simulation.cpp
+  ${SRC_DIR}/socks_server.cpp
+  ${SRC_DIR}/high_resolution_clock.cpp
+  ${SRC_DIR}/http_server.cpp
+  ${SRC_DIR}/queue.cpp
+  ${SRC_DIR}/simulator.cpp
+  ${SRC_DIR}/tcp_socket.cpp
+)
+
+add_library(simulator ${SIMULATOR_SRC_FILES})
```
Comment (arvidn) [inline CMakeLists.txt]: if you have a cmake project depend on this library, do you specify this target as a dependency? or is there some special command to define a target other projects can depend on? And, a project depending on this target, does it also get the include directories specified below?

File: CMakeLists.txt (hunk 1 above)
Comment (jberkel) [inline CMakeLists.txt]: no, you should use [CXX_STANDARD](https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html). it's just not supported in older cmakes (the one run by the linux travis instance), but that can be fixed with an env variable

File: CMakeLists.txt (hunk 2 above)
Comment (jberkel) [inline CMakeLists.txt]: This should work via [add_subdirectory](https://cmake.org/cmake/help/latest/command/add_subdirectory.html).

I don't think the include directories are propagated, but I'm not sure. The main reason I wanted to add cmake support is to make development on libsimulator itself easier (I find bjam confusing/limiting and some IDEs can easily auto-import the config from a `CMakeLists` file).

File: .travis.yml [hunk 3]
```diff
@@ -29,16 +29,18 @@ addons:
...
   - 'if [ $TRAVIS_OS_NAME == "osx" ]; then echo "using darwin : : ccache clang++ : <cflags>-std=c11 <cxxflags>-std=c++11 <compileflags>-Wno-deprecated-declarations ;" > ~/user-config.jam; fi'
-  - if [ $TRAVIS_OS_NAME == "osx" ]; then brew update > /dev/null && brew install --quiet ccache boost-build; fi
+  - if [ $TRAVIS_OS_NAME == "osx" ]; then brew update > /dev/null && brew install --quiet ccache boost-build && brew upgrade --quiet cmake ; fi
   - ccache -V && ccache --show-stats && ccache --zero-stats
 
 script:
 # until we have a version of boost-build that's new enough, disable warnings
 # as errors
   - bjam --hash -j3 variant=$variant -l900 warnings-as-errors=off $toolset $target
+  - mkdir build && cd build && cmake -DCMAKE_BUILD_TYPE=$variant .. && make -j3 && make test
```
Comment (arvidn) [inline .travis.yml]: I thought you had to say ``-G "Unix Makefiles"`` to ``cmake`` for it to generate makefiles.

Comment (jberkel) [inline .travis.yml]: on any unix-like system makefiles are probably generated by default

**Pull request merged**